application/models/Update_model.php

```diff
@@ -359,13 +359,87 @@ function tle() {
 
 		curl_close($curl);
 
-        $mtime = microtime();
-        $mtime = explode(" ",$mtime);
-        $mtime = $mtime[1] + $mtime[0];
-        $endtime = $mtime;
-        $totaltime = ($endtime - $starttime);
-        echo "This page was created in ".$totaltime." seconds <br />";
-        echo "Records inserted: " . $count . " <br/>";
+		$mtime = microtime();
+		$mtime = explode(" ",$mtime);
+		$mtime = $mtime[1] + $mtime[0];
+		$endtime = $mtime;
+		$totaltime = ($endtime - $starttime);
+		echo "This page was created in ".$totaltime." seconds <br />";
+		echo "Records inserted: " . $count . " <br/>";
 	}
 
+	 function lotw_sats() {
+		$url = 'https://lotw.arrl.org/lotw/config.tq6';
+		$curl = curl_init($url);
+
+		curl_setopt($curl, CURLOPT_FAILONERROR, true);
+		curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+		curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
+		curl_setopt($curl, CURLOPT_BINARYTRANSFER,true);
+		curl_setopt($curl, CURLOPT_TIMEOUT, 10);
+
+		$response = curl_exec($curl);
+		if (curl_errno($curl)) {
+			echo __('cURL error:').' '.curl_error($curl).' ('.curl_errno($curl).')';
+			return;
+		}
+		curl_close($curl);
+		$xml = simplexml_load_string(gzdecode($response));
```

Comment:
perhaps adding a try/catch around it?
don't know what happens if there's a 200 but with some kind of crap/empty body?